test/hotspot/jtreg/compiler/sharedstubs/SharedRuntimeCallTrampolineTest.java

```diff
@@ -72,38 +73,64 @@ private static void runTest(String compiler, String test) throws Exception {
     }
 
     public static void main(String[] args) throws Exception {
-        String[] tests = new String[] {"StaticMethodTest"};
+        String[] tests = new String[] { "RuntimeCallTest" };
         for (String test : tests) {
             runTest(args[0], test);
         }
     }
 
+    private static String getTestMethodStdout(OutputAnalyzer output) {
+        return Pattern.compile("Compiled method.*RuntimeCallTest::test").split(output.getStdout(), 2)[1];
+    }
+
     private static void checkOutput(OutputAnalyzer output) {
-        List<String> addrs = Pattern.compile("\\(trampoline_stub\\) addr=(\\w+) .*\\[trampoline owner")
-            .matcher(output.getStdout())
-            .results()
-            .map(m -> m.group(1))
-            .toList();
-        if (addrs.stream().distinct().count() >= addrs.size()) {
-            throw new RuntimeException("No runtime trampoline stubs reused: distinct " + addrs.stream().distinct().count() + ", in total " + addrs.size());
+        String testMethodStdout = getTestMethodStdout(output);
+        List<String> callAddrs = Pattern.compile("\\(runtime_call\\) addr=(\\w+) .*\\[destination")
+                .matcher(testMethodStdout)
+                .results()
+                .map(m -> m.group(1))
+                .toList();
+
+        record TrampolineReloc(String addr, String owner) {
         }
-    }
+        List<TrampolineReloc> trampolineRelocs = Pattern
+                .compile("\\(trampoline_stub\\) addr=(\\w+) .*\\[trampoline owner=(\\w+)]")
+                .matcher(testMethodStdout)
+                .results()
+                .map(m -> new TrampolineReloc(m.group(1), m.group(2)))
+                .toList();
 
-    public static class StaticMethodTest {
-        private static void log(int i, String msg) {
+        List<String> trampolineAddrs = trampolineRelocs.stream()
+                .filter(reloc -> callAddrs.contains(reloc.owner()))
+                .map(reloc -> new String(reloc.addr()))
+                .collect(Collectors.toList());
+        if (trampolineAddrs.stream().distinct().count() >= trampolineAddrs.size()) {
```

Comment:
For better readability, could you please create a meaningful variable for `trampolineAddrs.stream().distinct().count()`? You can reuse it in the exception message as well.